docs/resource/README.md

```diff
@@ -102,7 +102,105 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+
+If the instance has no inherent unique ID that can be used as the value of this attribute,
+implementations MAY generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. When enough data is available,
+implementations SHOULD use Version 5 and SHOULD use the following UUID as the namespace:
+`4d63009a-8d0f-11ee-aad7-4c796ed8e320`.
+
+UUIDs are typically recommended, as only an opaque yet reproducible value for
+the purposes of identifying a service instance is needed. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file,
+the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the input SHOULD be prefixed with
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}`, followed by the
+the workload identifier, which should tentatively be stable. The workload identifier is the final piece of
+information making the source of telemetry to be uniquely idenfiable. For instance, it might be the `container.name`
+in containerized or Kubernetes environments. This means that the same service yields the
+same UUID if the same identifier (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+SDKs SHOULD use the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this MUST take priority over generated IDs.
+- When the language, framework, or platform has a native mechanism allowing workloads to be uniquely idenfiable, like Erlang's
+  clustering identity, this mechanism MAY be used.
+- When any of the below combinations of resource attribute are provided, they MUST be used as the input
+  for generating a UUID v5 following the prefix mentioned above. The values within each combination MUST be separated with dots:
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`, resulting in the input
+  `${telemetry.sdk.name}.${telemetry.sdk.language}.${k8s.namespace.name}.${service.name}.${k8s.pod.name}.${k8s.container.name}`. In this case,
```

Comment:
```suggestion
  `${telemetry.sdk.name}.${telemetry.sdk.language}.${k8s.namespace.name}.${k8s.pod.name}.${k8s.container.name}`. In this case,
```

Reply:
You cannot have multiple services on a single container.

Reply:
That's true: a pod name is unique across a namespace. For completeness, I think a service.name would be good to have and would probably be less confusing on future iterations of this. Unless you have reasons to remove the service.name, I'd keep it for consistency.

Reply:
Yes, the reason is that the service.name may be missing - see [other comment](https://github.com/open-telemetry/semantic-conventions/pull/312#discussion_r1486488318)

Reply:
OTOH, I would propose to allow for service name too

- k8s + service name -> java application server that serves multiple SDKs on the same pod (doesn't exist so far, but would make sense)
- k8s without service name -> service name might be missing, but it's good enough to identify service.instance.id

Reply:
> k8s + service name -> java application server that serves multiple SDKs on the same pod (doesn't exist so far, but would make sense)

Don't we support tomcat with autoinstrumentation?

Reply:
Yes, but you can't have have different service names for the deployed apps inside tomcat.